In [1]:
# Imports
import sklearn as sk
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import scipy.stats as sp
import seaborn as sb

from sklearn.cluster import DBSCAN

In [2]:
# Cargar dataset
countries = pd.read_csv('country_data.csv')
print(countries)
np.shape(countries)[1]
countries.columns

                 country  child_mort  exports  health  imports  income  \
0            Afghanistan        90.2     10.0    7.58     44.9    1610   
1                Albania        16.6     28.0    6.55     48.6    9930   
2                Algeria        27.3     38.4    4.17     31.4   12900   
3                 Angola       119.0     62.3    2.85     42.9    5900   
4    Antigua and Barbuda        10.3     45.5    6.03     58.9   19100   
..                   ...         ...      ...     ...      ...     ...   
162              Vanuatu        29.2     46.6    5.25     52.7    2950   
163            Venezuela        17.1     28.5    4.91     17.6   16500   
164              Vietnam        23.3     72.0    6.84     80.2    4490   
165                Yemen        56.3     30.0    5.18     34.4    4480   
166               Zambia        83.1     37.0    5.89     30.9    3280   

     inflation  life_expec  total_fer   gdpp  
0         9.44        56.2       5.82    553  
1         4.49   

Index(['country', 'child_mort', 'exports', 'health', 'imports', 'income',
       'inflation', 'life_expec', 'total_fer', 'gdpp'],
      dtype='str')

In [3]:
# Estandardizar datos
dataframe = countries.iloc[:, 0:10]
dataframe[dataframe.columns[1:10]] = dataframe[dataframe.columns[1:10]].astype(float)

dataframe[dataframe.columns[1:10]] = sk.preprocessing.StandardScaler().fit_transform(dataframe[dataframe.columns[1:10]])

print(np.mean(dataframe.iloc[:,1]))
print(np.std(dataframe.iloc[:,1]))
print(dataframe)

-3.722903555629267e-17
1.0
                 country  child_mort   exports    health   imports    income  \
0            Afghanistan    1.291532 -1.138280  0.279088 -0.082455 -0.808245   
1                Albania   -0.538949 -0.479658 -0.097016  0.070837 -0.375369   
2                Algeria   -0.272833 -0.099122 -0.966073 -0.641762 -0.220844   
3                 Angola    2.007808  0.775381 -1.448071 -0.165315 -0.585043   
4    Antigua and Barbuda   -0.695634  0.160668 -0.286894  0.497568  0.101732   
..                   ...         ...       ...       ...       ...       ...   
162              Vanuatu   -0.225578  0.200917 -0.571711  0.240700 -0.738527   
163            Venezuela   -0.526514 -0.461363 -0.695862 -1.213499 -0.033542   
164              Vietnam   -0.372315  1.130305  0.008877  1.380030 -0.658404   
165                Yemen    0.448417 -0.406478 -0.597272 -0.517472 -0.658924   
166               Zambia    1.114951 -0.150348 -0.338015 -0.662477 -0.721358   

     inflati

In [24]:
epsilons = [0.5, 0.75, 1]
min_points = [2,4,6]

for e in epsilons:
    for p in min_points:

        clustering = DBSCAN(eps=e, min_samples=p).fit(dataframe[dataframe.columns[1:10]])
        labels = clustering.labels_
        print(f"\n DBSCAN epsilon: {e}, min_points: {p}\n", labels)
        wcss = 0
        unique_labels = set(labels)
        for label in unique_labels:
            if label == -1:  # No tiene sentido calcular el centroide para los puntos etiquetados como ruido
                continue
            
            cluster_points = dataframe[dataframe.columns[1:10]][labels == label]
            
            centroid = np.mean(cluster_points, axis=0)
            
            wcss += np.sum((cluster_points - centroid) ** 2)
        anomalies = dataframe[labels == -1]
        print("Anomalías:", len(anomalies))
        
        '''
        for country, cluster in zip(dataframe.iloc[:, 0], labels):
                print(f"{country}: cluster {cluster}")
        '''
        print("Inercia", wcss)
        print("Numero clusteres:",len(unique_labels)-1 ) # -1 para no contar el cluster de ruido


 DBSCAN epsilon: 0.5, min_points: 2
 [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  2 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1  2 -1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  2 -1 -1 -1 -1 -1 -1 -1 -1
 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  2 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  2 -1 -1 -1 -1 -1 -1 -1 -1]
Anomalías: 158
Inercia 0.9533521388950206
Numero clusteres: 3

 DBSCAN epsilon: 0.5, min_points: 4
 [-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 

Considerando el valor de la inercia como criterio de elección, excluyendos los casos de inercia cero, el mejor algoritmo sería el que utiliza un épsilon de 0.5 y un min_points de 2. En este caso, obtendríamos 158 anomalías (ruido) y solo 3 clústeres. 

In [26]:
epsilons = [1, 1.5, 2, 2.5, 3]
min_points = [2,4,6]

for e in epsilons:
    for p in min_points:

        clustering = DBSCAN(eps=e, min_samples=p).fit(dataframe[dataframe.columns[1:10]])
        labels = clustering.labels_
        print(f"\n DBSCAN epsilon: {e}, min_points: {p}\n", labels)
        wcss = 0
        unique_labels = set(labels)
        for label in unique_labels:
            if label == -1:  # No tiene sentido calcular el centroide para los puntos etiquetados como ruido
                continue
            
            cluster_points = dataframe[dataframe.columns[1:10]][labels == label]
            
            centroid = np.mean(cluster_points, axis=0)
            
            wcss += np.sum((cluster_points - centroid) ** 2)
        anomalies = dataframe[labels == -1]
        print("Anomalías:", len(anomalies))
        
        '''
        for country, cluster in zip(dataframe.iloc[:, 0], labels):
                print(f"{country}: cluster {cluster}")
        '''
        print("Inercia", wcss)
        print("Numero clusteres:",len(unique_labels)-1 ) # -1 para no contar el cluster de ruido


 DBSCAN epsilon: 1, min_points: 2
 [ 0  1 -1 -1  1 -1  1  2  2 -1  1 -1  3  1 -1  4  1  5  1  1  1 -1  1  6
  1  5 -1  1  5  2  1 -1  7  1  1  1  8 -1 -1  1  0  1  1  1  2  1  1  1
  1 -1 -1  1  1  2  2 -1  8  1  2 -1  2  1  1  0  5  1 -1  1  2  3 -1  1
 -1 -1 -1  2  1  2 -1 -1  8 -1  6  1 -1  1  1 -1 -1 -1  1 -1  1  8  0 -1
  1  7 -1 -1  1 -1 -1 -1  1  1  0 -1 -1  9  4  2  7 -1 -1 10 -1  1  1  1
  1  1  2 -1  1  1 -1 11 10  8  1 -1 -1 -1  1  1 -1 -1  1  2 -1  1 -1  1
  2 -1 -1  0 -1 -1 -1 11  1  1 -1  0 -1 -1  2 -1  1  9  1 -1 -1 -1  0]
Anomalías: 60
Inercia 147.01039941174992
Numero clusteres: 12

 DBSCAN epsilon: 1, min_points: 4
 [ 2  0 -1 -1  0 -1  0  1  1 -1  0 -1 -1  0 -1 -1  0 -1  0  0  0 -1  0 -1
  0 -1 -1  0 -1  1  0 -1 -1  0  0  0  3 -1 -1  0  2  0 -1  0  1  0  0  0
  0 -1 -1  0  0  1  1 -1  3  0  1 -1  1  0  0  2 -1  0 -1  0  1 -1 -1  0
 -1 -1 -1  1  0  1 -1 -1  3 -1 -1  0 -1  0  0 -1 -1 -1  0 -1  0 -1  2 -1
  0 -1 -1 -1  0 -1 -1 -1  0  0  2 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -

Sin considerar la inercia, para elegir los mejores hiperparámetros tuve que realizar una búsqueda manual. Los parámetros min_points y epsilon modifican cómo se agrupan los clústeres. Al probar con varios valores, noté que con un épsilon menor a 1 los clústeres no se unían (lo que provocaba que demasiados países fueran etiquetados como anomalías). Por ello, probé con valores más altos, buscando un equilibrio para evitar la creación de un único "mega clúster".

Los valores óptimos son aquellos que no agrupan todos los puntos en un mismo clúster (0) pero que tampoco generan un exceso de anomalías. Los parámetros que mejor se ajustan a esta descripción son:

    Epsilon 1, min_points 4: Resulta en 85 anomalías y 4 clústeres.

    Epsilon 1.5, min_points 2: Resulta en 23 anomalías y 3 clústeres.

Para elegir el mejor entre ambos, procedo a analizar las etiquetas asignadas a cada país.

In [5]:
clustering = DBSCAN(eps=1, min_samples=4).fit(dataframe[dataframe.columns[1:10]])
labels = clustering.labels_

for country, cluster in zip(dataframe.iloc[:, 0], labels):
    print(f"{country}: cluster {cluster}")

Afghanistan: cluster 2
Albania: cluster 0
Algeria: cluster -1
Angola: cluster -1
Antigua and Barbuda: cluster 0
Argentina: cluster -1
Armenia: cluster 0
Australia: cluster 1
Austria: cluster 1
Azerbaijan: cluster -1
Bahamas: cluster 0
Bahrain: cluster -1
Bangladesh: cluster -1
Barbados: cluster 0
Belarus: cluster -1
Belgium: cluster -1
Belize: cluster 0
Benin: cluster -1
Bhutan: cluster 0
Bolivia: cluster 0
Bosnia and Herzegovina: cluster 0
Botswana: cluster -1
Brazil: cluster 0
Brunei: cluster -1
Bulgaria: cluster 0
Burkina Faso: cluster -1
Burundi: cluster -1
Cambodia: cluster 0
Cameroon: cluster -1
Canada: cluster 1
Cape Verde: cluster 0
Central African Republic: cluster -1
Chad: cluster -1
Chile: cluster 0
China: cluster 0
Colombia: cluster 0
Comoros: cluster 3
Congo, Dem. Rep.: cluster -1
Congo, Rep.: cluster -1
Costa Rica: cluster 0
Cote d'Ivoire: cluster 2
Croatia: cluster 0
Cyprus: cluster -1
Czech Republic: cluster 0
Denmark: cluster 1
Dominican Republic: cluster 0
Ecuador: cl

In [6]:
clustering = DBSCAN(eps=1.5, min_samples=2).fit(dataframe[dataframe.columns[1:10]])
labels = clustering.labels_

for country, cluster in zip(dataframe.iloc[:, 0], labels):
    print(f"{country}: cluster {cluster}")

Afghanistan: cluster 0
Albania: cluster 0
Algeria: cluster 0
Angola: cluster -1
Antigua and Barbuda: cluster 0
Argentina: cluster 0
Armenia: cluster 0
Australia: cluster 0
Austria: cluster 0
Azerbaijan: cluster 0
Bahamas: cluster 0
Bahrain: cluster 0
Bangladesh: cluster 0
Barbados: cluster 0
Belarus: cluster 0
Belgium: cluster 0
Belize: cluster 0
Benin: cluster 0
Bhutan: cluster 0
Bolivia: cluster 0
Bosnia and Herzegovina: cluster 0
Botswana: cluster 0
Brazil: cluster 0
Brunei: cluster 1
Bulgaria: cluster 0
Burkina Faso: cluster 0
Burundi: cluster 0
Cambodia: cluster 0
Cameroon: cluster 0
Canada: cluster 0
Cape Verde: cluster 0
Central African Republic: cluster -1
Chad: cluster 0
Chile: cluster 0
China: cluster 0
Colombia: cluster 0
Comoros: cluster 0
Congo, Dem. Rep.: cluster -1
Congo, Rep.: cluster -1
Costa Rica: cluster 0
Cote d'Ivoire: cluster 0
Croatia: cluster 0
Cyprus: cluster 0
Czech Republic: cluster 0
Denmark: cluster 0
Dominican Republic: cluster 0
Ecuador: cluster 0
Egypt: 

Entre estos dos, considero que el mejor es epsilon 1, min_points 4, ya que el otro une países que son demasiado diferentes entre sí. Con esta configuración obtenemos un clúster más y un mayor número de anomalías; esto indica que el algoritmo DBSCAN está siendo más selectivo, aunque no logremos agrupar una gran cantidad de países en un solo clúster.

Sin embargo desde ahora usaremos los parametros calculados antes, aquellos que tienen mejor inercia.

In [7]:
# Silhoutte score
from sklearn.metrics import silhouette_score
# Kmeans
from sklearn.cluster import KMeans
kmeans = KMeans(n_clusters=3, random_state=42)
kmeans.fit(dataframe.iloc[:, 1:10])
print(kmeans.inertia_)
clusters = kmeans.fit_predict(dataframe.iloc[:,1:10])

score = silhouette_score(dataframe.iloc[:,1:10], clusters)
print(f"Silhouette Score: {score}")

for country, cluster in zip(dataframe.iloc[:, 0], clusters):
    print(f"{country}: cluster {cluster}")

831.5351582151586
Silhouette Score: 0.285600988953231
Afghanistan: cluster 2
Albania: cluster 0
Algeria: cluster 0
Angola: cluster 2
Antigua and Barbuda: cluster 0
Argentina: cluster 0
Armenia: cluster 0
Australia: cluster 1
Austria: cluster 1
Azerbaijan: cluster 0
Bahamas: cluster 0
Bahrain: cluster 1
Bangladesh: cluster 0
Barbados: cluster 0
Belarus: cluster 0
Belgium: cluster 1
Belize: cluster 0
Benin: cluster 2
Bhutan: cluster 0
Bolivia: cluster 0
Bosnia and Herzegovina: cluster 0
Botswana: cluster 0
Brazil: cluster 0
Brunei: cluster 1
Bulgaria: cluster 0
Burkina Faso: cluster 2
Burundi: cluster 2
Cambodia: cluster 0
Cameroon: cluster 2
Canada: cluster 1
Cape Verde: cluster 0
Central African Republic: cluster 2
Chad: cluster 2
Chile: cluster 0
China: cluster 0
Colombia: cluster 0
Comoros: cluster 2
Congo, Dem. Rep.: cluster 2
Congo, Rep.: cluster 2
Costa Rica: cluster 0
Cote d'Ivoire: cluster 2
Croatia: cluster 0
Cyprus: cluster 1
Czech Republic: cluster 1
Denmark: cluster 1
Domini

In [8]:
# Agglomerative Clustering
from sklearn.cluster import AgglomerativeClustering
hc = AgglomerativeClustering(n_clusters=3, linkage='ward')
clusters_hc = hc.fit_predict(dataframe.iloc[:, 1:10])
score = silhouette_score(dataframe.iloc[:,1:10], clusters_hc)

print(f"Silhouette Score: {score}")

for country, cluster in zip(dataframe.iloc[:, 0], clusters_hc):
    print(f"{country}: cluster {cluster}")

Silhouette Score: 0.24563001303300652
Afghanistan: cluster 2
Albania: cluster 1
Algeria: cluster 1
Angola: cluster 1
Antigua and Barbuda: cluster 1
Argentina: cluster 1
Armenia: cluster 1
Australia: cluster 0
Austria: cluster 0
Azerbaijan: cluster 1
Bahamas: cluster 1
Bahrain: cluster 0
Bangladesh: cluster 1
Barbados: cluster 1
Belarus: cluster 1
Belgium: cluster 0
Belize: cluster 1
Benin: cluster 2
Bhutan: cluster 1
Bolivia: cluster 1
Bosnia and Herzegovina: cluster 1
Botswana: cluster 1
Brazil: cluster 1
Brunei: cluster 0
Bulgaria: cluster 1
Burkina Faso: cluster 2
Burundi: cluster 2
Cambodia: cluster 1
Cameroon: cluster 2
Canada: cluster 0
Cape Verde: cluster 1
Central African Republic: cluster 2
Chad: cluster 2
Chile: cluster 1
China: cluster 1
Colombia: cluster 1
Comoros: cluster 2
Congo, Dem. Rep.: cluster 2
Congo, Rep.: cluster 1
Costa Rica: cluster 1
Cote d'Ivoire: cluster 2
Croatia: cluster 1
Cyprus: cluster 1
Czech Republic: cluster 1
Denmark: cluster 0
Dominican Republic: cl

In [27]:
# DBSCAN 
# Los iperparametros elegidos son los que han dado mejor inercia, como pedido por el enunciado, aunque no sea el mejor resultado
clustering = DBSCAN(eps=0.5, min_samples=2).fit(dataframe[dataframe.columns[1:10]])
labels = clustering.labels_
score = silhouette_score(dataframe.iloc[:,1:10], labels)
print(f"Silhouette Score: {score}")
for country, cluster in zip(dataframe.iloc[:, 0], labels):
    print(f"{country}: cluster {cluster}")

Silhouette Score: -0.2535351890412684
Afghanistan: cluster -1
Albania: cluster -1
Algeria: cluster -1
Angola: cluster -1
Antigua and Barbuda: cluster -1
Argentina: cluster -1
Armenia: cluster -1
Australia: cluster -1
Austria: cluster -1
Azerbaijan: cluster -1
Bahamas: cluster -1
Bahrain: cluster -1
Bangladesh: cluster -1
Barbados: cluster -1
Belarus: cluster -1
Belgium: cluster -1
Belize: cluster -1
Benin: cluster -1
Bhutan: cluster -1
Bolivia: cluster -1
Bosnia and Herzegovina: cluster -1
Botswana: cluster -1
Brazil: cluster -1
Brunei: cluster -1
Bulgaria: cluster 0
Burkina Faso: cluster -1
Burundi: cluster -1
Cambodia: cluster -1
Cameroon: cluster -1
Canada: cluster -1
Cape Verde: cluster -1
Central African Republic: cluster -1
Chad: cluster -1
Chile: cluster -1
China: cluster -1
Colombia: cluster -1
Comoros: cluster -1
Congo, Dem. Rep.: cluster -1
Congo, Rep.: cluster -1
Costa Rica: cluster -1
Cote d'Ivoire: cluster -1
Croatia: cluster 1
Cyprus: cluster -1
Czech Republic: cluster -1

Por lo tanto, el mejor modelo según el Silhouette Score es K-Means, ya que obtuvo un valor ligeramente superior al del Clustering Aglomerativo. Por el contrario, el resultado de DBSCAN, al estar más abajo de cero, resulta prácticamente erróneo para este conjunto de datos.

De hecho, según la escala que hemos visto en clase, una puntuación superior a 0,25 indica que 'algunos puntos están mal agrupados o cerca de los límites'; esto significa que, aunque no hemos obtenido una clasificación perfecta, es el mejor resultado que podíamos alcanzar con estos parámetros.

In [12]:
# KMeans analisis
kmeans = KMeans(n_clusters=3, random_state=42)
kmeans.fit(dataframe.iloc[:, 1:10])
clusters = kmeans.predict(dataframe.iloc[:,1:10])
# Por cada cluster calculamos 
# child_mort
for i in range(3):
    cluster_data = dataframe.iloc[:, 1:10][clusters == i]
    mean_child_mort = cluster_data['child_mort'].mean()
    print(f"Cluster {i}: Media child_mort = {mean_child_mort}")
# exports
for i in range(3):
    cluster_data = dataframe.iloc[:, 1:10][clusters == i]
    mean_exports = cluster_data['exports'].mean()
    print(f"Cluster {i}: Media exports = {mean_exports}")
# income
for i in range(3):
    cluster_data = dataframe.iloc[:, 1:10][clusters == i]
    mean_income = cluster_data['income'].mean()
    print(f"Cluster {i}: Media income = {mean_income}")
# total_fer
for i in range(3):
    cluster_data = dataframe.iloc[:, 1:10][clusters == i]
    mean_total_fer = cluster_data['total_fer'].mean()
    print(f"Cluster {i}: Media total_fer = {mean_total_fer}")
# mean child_mort
print(f"\nMedia child_mort = {dataframe['child_mort'].mean()}")
# mean exports
print(f"Media exports = {dataframe['exports'].mean()}")
# mean income
print(f"Media income = {dataframe['income'].mean()}")
# mean total_fer
print(f"Media total_fer = {dataframe['total_fer'].mean()}")

Cluster 0: Media child_mort = -0.3932819644433522
Cluster 1: Media child_mort = -0.8274486635067491
Cluster 2: Media child_mort = 1.4135644628526944
Cluster 0: Media exports = -0.030583746766141416
Cluster 1: Media exports = 0.6450798457714426
Cluster 2: Media exports = -0.4576149383529718
Cluster 0: Media income = -0.25093020946788935
Cluster 1: Media income = 1.4842426811375093
Cluster 2: Media income = -0.707838633482486
Cluster 0: Media total_fer = -0.4025786187815343
Cluster 1: Media total_fer = -0.7918768655237743
Cluster 2: Media total_fer = 1.402873963868175

Media child_mort = -3.722903555629267e-17
Media exports = 2.1273734603595814e-16
Media income = -7.97765047634843e-17
Media total_fer = 3.044803265139651e-16


In [23]:
import plotly.express as px
import pandas as pd

dataframe['cluster'] = clusters.astype(str)
dataframe['cluster_label'] = dataframe['cluster'].map({ # Nombre clusters
    '0': 'Cluster 0',
    '1': 'Cluster 1',
    '2': 'Cluster 2'
})

# Sunburst plot
fig = px.sunburst(
    dataframe,
    path=['cluster_label', 'country'],  
    title='Sunburst Plot - Paesi per Cluster (KMeans)',
    color='cluster_label',
    color_discrete_sequence=px.colors.qualitative.Set2
)
plt.figure(figsize=(50, 40))

fig.update_layout(margin=dict(t=50, l=0, r=0, b=0))
fig.show()

<Figure size 5000x4000 with 0 Axes>

A traves de esta imagen se nota como el Cluster 0 es el mas largo que todos. Mas que la mitad del total de los paises partenece a esto. Y che muy pocos paises partenecen al Cluster 1. 

Basándonos en las variables podemos identificar tres perfiles socioeconómicos claramente diferenciados:

1. Cluster 2: Países en Situación de Pobreza

Este grupo presenta los indicadores más críticos. Se caracteriza por una mortalidad infantil muy superior a la media (1.41) y una tasa de fertilidad elevada (1.40). Estos datos, sumados a los niveles más bajos de ingresos (-0.71) y exportaciones (-0.46), sugieren economías en desarrollo con serias carencias en infraestructura de salud y baja capacidad de hacer compras.

2. Cluster 1: Países Desarrollados

Es el polo opuesto al Cluster 2. Muestra los niveles de ingresos más altos (1.48) y una fuerte actividad comercial (exports = 0.65). Socialmente, son países muy estables, con la mortalidad infantil más baja (-0.83) y tasas de fertilidad por debajo del promedio global. Representa el perfil de naciones industrializadas con alta calidad de vida.

3. Cluster 0: Países en via de Desarrollo (el cluster mas largo)

Este grupo se sitúa en un terreno intermedio, aunque con unos indicadores positivos. Sus niveles de mortalidad infantil (-0.39) y fertilidad (-0.40) están por debajo de la media global, lo que indica avances significativos en salud y desarrollo social. Sus ingresos y exportaciones están cerca del promedio, representando probablemente a economías emergentes o países en vías de desarrollo.

Se confirman tambien las analises hechas por la primera parte del proyecto. 